src/main/java/com/example/springbootbookshop/repository/BookRepository.java

```diff
@@ -1,9 +1,14 @@
 package com.example.springbootbookshop.repository;
 
 import com.example.springbootbookshop.entity.Book;
+import java.util.List;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.stereotype.Repository;
 
 @Repository
 public interface BookRepository extends JpaRepository<Book, Long> {
+
+    @Query("SELECT b FROM Book b JOIN b.categories c WHERE c.id = :categoryId")
```

Comment:
if you need to fetch books along with category - use `join fetch`

if not - feel free to skip this comment